### ISSUE TITLE
You must be a logged-in, registered member of this site to perform this action.

### ISSUE THREAD

**Issue by PatPeter** [xfrocks/bdApi#38]
While logged into XenForo, I try to run the following admin commands and it says I'm not logged in.

```
GET /api/index.php?users/me

Request
array(0) {
}
Response (403)
array(1) {
  ["error"] => string(34) "You must be a logged-in, registered member of this site to perform this action. "
}
```

How can I connect the API with the forums so that it can read authentication tokens?

I am trying to implement the [XenForoAuth MediaWiki extension](https://www.mediawiki.org/wiki/Extension:XenForoAuth).

**Comment by daohoangson** [comment]
In general, the API endpoints do recognize browser session so you need to use token via `oauth_token` parameter. For testing purposes, you can create one in AdminCP > Tools > [bd] API > Tokens.

**Comment by PatPeter** [comment]
Alright, good news! I was able to access my user profile data by providing an oauth_token.

`/api/index.php?users/me&oauth_token=asdf1234`

However, creating the token doesn't fix the wiki's inability to login. It seems that it's been an issue [since [bd] API released 1.5.1](https://www.mediawiki.org/wiki/Extension_talk:XenForoAuth).

Is there a way to request an OAuth Token given only a client ID / client secret in order to pass to `users/me`?

**Comment by daohoangson** [comment]
You will need to check the MediaWiki plugin to see what it is expecting from the API.

In version 1.5.1 of [bd] API, there are not that many changes especially in authentication for a basic endpoint like /users/me. It has required a valid token from the beginning.
